stdlib/LibGit2/src/callbacks.jl

```diff
@@ -359,9 +359,29 @@ function fetchhead_foreach_callback(ref_name::Cstring, remote_url::Cstring,
     return Cint(0)
 end
 
+# from `enum git_cert_t` in `cert.h`
+const CERT_TYPE_TLS = 1 # GIT_CERT_X509
+const CERT_TYPE_SSH = 2 # GIT_CERT_HOSTKEY_LIBSSH2
+
+function certificate_callback(
+    cert_p :: Ptr{Cvoid},
+    valid  :: Cint,
+    host_p :: Ptr{Cchar},
+    user_p :: Ptr{Cvoid},
+)::Cint
+    host = unsafe_string(host_p)
+    cert_type = unsafe_load(convert(Ptr{Cint}, cert_p))
+    transport = cert_type == CERT_TYPE_TLS ? "TLS" :
+                cert_type == CERT_TYPE_SSH ? "SSH" : nothing
+    @show host, transport
```

Comment:
Don't forget to remove this

Reply:
Will do. I'm still testing here. Should have marked the PR as draft for now.